cmd/root.go

```diff
@@ -51,6 +51,8 @@ func Execute() {
 	viper.BindPFlag("verbosity", pf.Lookup("verbosity"))
 	argToEnv["verbosity"] = "VERBOSITY"
 
+	client.CmdPush.Flags().StringVarP(&client.FlagServe, "serve", "s", "", "inference service to serve the model (kfserving, knative, deployment)")
```

Comment:
I think the `deployment` option is confusing. Actually `knative` too. They both serve through mlflow so the serve option should really be `mlflow`. If you still want to expose the option to use knative/non-knative way, it should be another argument.

Reply:
I partially agree, I do not think that mlflow is at the same level as kfserving/seldon. For example, if I choose kfserving the serving is done through mlserver and if I choose seldon the serving can also be done with mlflow or other tool.

In that way I think deployment/knative as the kind of serving is more close to be in the same level as kfserving/seldon.

Reply:
Hm, I think it will be confusing for users. But we can leave this discussion for later.

Reply:
My 2c: there's an option to choose the tool used for serving: `knative`, `kfserving`, `seldon` or `plain kubernetes` (deployment), and there's a sub-option specific to each tool used to choose the serving prediction mechanism. The sub-options could be (for now): `mlflow` (use mlflow instead of a builtin prediction server), `detect` (detect the predictor type based on the library used to train/save the model) or enforce one of the supported libraries (sklearn, pytorch, tensorflow, etc).

Reply:
My suggestions is based on the assumption that we will support tool specific recipes that users can choose from to implement a particular MLOps segment. In this case, these are all serving segments: `knative`, `kfserving`, `seldon` etc. Then there are some generic parameters that all serving segments must support (e.g. the model URL, the prediction engine). And some particular parameters that are specific to each tool.